speechbrain/lobes/models/huggingface_transformers/huggingface.py

```diff
@@ -0,0 +1,404 @@
+        is hosted locally or on a HuggingFace hub.
+        Called as static function in HFTransformersInterface._from_pretrained.
+        Arguments
+        ---------
+        path : str
+            Used as "source"; local path or HuggingFace hub name: e.g "facebook/wav2vec2-large-lv60"
+        save_path : str
+            norm_output (dir) of the downloaded model.
+        Returns
+        -------
+        is_sb : bool
+            Whether/not the model is deserializable w/ SpeechBrain or not (then, model conversion is needed).
+        checkpoint_filename : str
+            as of HuggingFace documentation: file name relative to the repo root (guaranteed to be here).
+        """
+        checkpoint_filename = ""
+        source = pathlib.Path(path)
+        is_local = True
+
+        # If path is a huggingface hub.
+        if not source.exists():
+            is_local = False
+
+        # Check if source is downloaded already
+        sink = pathlib.Path(
+            save_path + "/models--" + path.replace("/", "--") + "/snapshots"
+        )
+        if sink.exists():
+            sink = (
+                sink / os.listdir(str(sink))[0]
+            )  # there's a hash-id subfolder
+            if any(
+                File.endswith(".bin") or File.endswith(".ckpt")
+                for File in os.listdir(str(sink))
+            ):
+                is_local = True
+                local_path = str(sink)
+            else:
+                local_path = path
+        else:
+            local_path = path
+
+        if is_local:
+            # Test for HuggingFace model
+            if any(File.endswith(".bin") for File in os.listdir(local_path)):
+                is_sb = False
+                return is_sb, checkpoint_filename, is_local
+
+            # Test for SpeechBrain model and get the filename.
+            for File in os.listdir(local_path):
+                if File.endswith(".ckpt"):
+                    checkpoint_filename = os.path.join(path, File)
+                    is_sb = True
+                    return is_sb, checkpoint_filename, is_local
+        else:
+            files = model_info(
+                path
+            ).siblings  # get the list of files of the Hub
+
+            # Test if it's an HuggingFace model or a SB one
+            for File in files:
+                if File.rfilename.endswith(".ckpt"):
+                    checkpoint_filename = File.rfilename
+                    is_sb = True
+                    return is_sb, checkpoint_filename, is_local
+
+            for File in files:
+                if File.rfilename.endswith(".bin"):
+                    checkpoint_filename = File.rfilename
+                    is_sb = False
+                    return is_sb, checkpoint_filename, is_local
+
+        err_msg = f"{path} does not contain a .bin or .ckpt checkpoint !"
+        raise FileNotFoundError(err_msg)
+
+    def _modify_state_dict(self, path, **args):
```

Comment:
`**args` -> `**kwargs` + put it in the docstring by adding a small line explaining on which case it might be useful to give more parameters